cmd/thanos/compact.go

```diff
@@ -281,14 +285,12 @@ func runCompact(
 
 			return errors.Wrap(err, "error executing compaction")
 		})
-	}, func(error) {
+	}, func(err error) {
+		readinessProber.SetNotReady(err)
 		cancel()
 	})
 
-	if err := metricHTTPListenGroup(g, logger, reg, httpBindAddr); err != nil {
-		return err
-	}
-
 	level.Info(logger).Log("msg", "starting compact node")
+	readinessProber.SetReady()
```

Comment:
This is not true, as we are not rdy at this point. We are rdy once our HTTP listener is able to accept new connections ):

Reply:
@FUSAKLA what about this comment? It seems like this still hasn't been addressed.

Reply:
No, it was removed at one point but then added back again eventually.

There is the issue that if you have sidecar it uses the `defaultHTTPListener` which should have set it healthy and ready right away. But then the sidecar is not ready when it cannot get the labels from prometheus instance so from start is should not be ready and wait till the first successful labels fetch. 

So I'm not sure how to do this. I need these two conditions to be met at once to say that the sidecar is ready right? 

There were two variants of having each component handling it's ready status on it's own or move setting of the ready status to the generic `defaultHTTPListener`. 

WDYT?